include/openssl/ec.h

```diff
@@ -1548,6 +1550,7 @@ OSSL_DEPRECATEDIN_3_0 void EC_KEY_METHOD_get_verify
 
 #  define EVP_EC_gen(curve) \
     EVP_PKEY_Q_keygen(NULL, NULL, "EC", (char *)(strstr(curve, "")))
+    /* strstr is used to enable type checking for the variadic string arg */
```

Comment:
How about: _strstr is used to enable type checking for the `curve' argument_

The term _variadic string arg_ is kind of non-specific when the macro argument has a name.

What about a separate `char *ossl_check_arg_is_string(char *arg)` function that wraps this for us.

Reply:
We have similar tricks in the safestack code. E.g. see:

https://github.com/openssl/openssl/blob/6be83cc655af819be0e3f2701c726a2550357953/include/openssl/safestack.h.in#L37-L65

I'm not sure if its ok to introduce such a trick for 3.0 though.

Reply:
???? wWhat tricks ????

Inline functions aren't a great solution here since they'll potentially still be instantiated when the macro isn't used.

Reply:
> ???? wWhat tricks ????

We have inline functions which only have the purpose of checking the type in a macro.

> Inline functions aren't a great solution here since they'll potentially still be instantiated when the macro isn't used.

So?

Reply:
Given how many problems we've had with inline functions defined in header files, I'm very hesitant to add another one.

The safestack.h functions are part of a macro which only defines them if they are being used.  This is a very significant difference.

Reply:
> The safestack.h functions are part of a macro which only defines them if they are being used.

This is not the case. They are defined unconditionally. IIRC the problems we had previously were with inline functions that contained references to symbols that weren't then linked. These inline functions have no references to other symbols.

Reply:
The point was that they produce no code **unless** the macro is used.

This is different to using inline functions here where the function will be defined even if the macro isn't used.